### PR TITLE
PLT-7263: truncate long issue descriptions

### PR DESCRIPTION
**Pull request by ccbrown**
This truncates long messages, then appends an ellipsis ~and another issue link~.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7263

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by jasonblais** [comment]
@ccbrown Looks like the build failed, can you help take a look? I'll spin up a test server afterwards.

Is this largely updates to test data and unit tests, apart from the four lines of code here? https://github.com/mattermost/platform/pull/7142/files#diff-bca816af108ea9f9ea7db9cade44453c

**Comment by ccbrown** [comment]
@jasonblais Right, I forgot that we hadn't merged https://github.com/mattermost/platform/pull/7116 yet, which is a dependency of the tests. I'll remove them from this PR so we can get this in first.

**Comment by jasonblais** [comment]
Sounds good.

For long descriptions, it looks like the ticket title gets added at the end for a second time. Otherwise looks good to me

image

**Comment by ccbrown** [comment]
@jasonblais Yeah, that _was_ intentional, but now on second thought is pretty weird... Removed!

Gonna go ahead and tag this for dev review then. @grundleborg, @jwilander you guys are up!

File: app/plugin/jira/webhook.go [hunk 1]
```diff
@@ -92,7 +92,11 @@ func (w *Webhook) SlackAttachment() (*model.SlackAttachment, error) {
 	text, err := w.renderText("" +
 		"[{{.Issue.Fields.Summary}}]({{.JIRAURL}}/browse/{{.Issue.Key}})" +
 		"{{if eq .WebhookEvent \"jira:issue_created\"}}{{if ne .Issue.Fields.Description \"\"}}" +
+		"{{if len .Issue.Fields.Description | lt 3000}}" +
```
Comment (grundleborg) [inline app/plugin/jira/webhook.go]: Clearly this works, as JB has tested its functionality, but I'm reading this line as `if len(description < 3000`, which would print the ellipsis for *shorter* messages, not longer ones. What am I missing?

Comment (ccbrown) [inline app/plugin/jira/webhook.go]: Pipelines pass their input as the _last_ argument in the next function. It's been bothering me too how this reads unintuitively.

Comment (grundleborg) [inline app/plugin/jira/webhook.go]: ahhh thanks. so it's `if 3000 < len(description)` - that makes sense.

**Pull request merged**